File: cpp 2025/nums_between.cpp
#include <iostream>
using namespace std;

// Code made in 04/12/2025

int main_code();

//Returns the amount of numbers between start & end
int nums_in(int s, int e){
    int nums_between = 0;
    for(int i = s+1; i < e; i++){nums_between += 1;}

    return nums_between;
}

//Returns the sum of them all
int sum(int s, int e){
    int total = 0;
    for(int i = s+1; i < e; i++){total += i;}
    return total;
}

//Returns the ammount of negative numbers
int negatives_num(int s, int e){
    int count = 0;
    for(int i = s+1; i < e; i++){
        if(i < 0){count++;}
    }
    return count;
}

//Returns the ammount of positive numbers
int positive_nums(int s, int e){
    int count = 0;
    for(int i = s+1; i < e; i++){
        if(i > 0){count++;}
    }
    return count;
}

//Returns the ammount of evens
int even(int s, int e){
    int evens = 0;
    for(int i = s+1; i < e; i++){
        if (i%2 == 0){evens++;}
    }
    return evens;
}


// Main
int main (){
    cout << "Digit a start and an end to know about the numbers between them!\n";
    cout << "Numbers must be at least 2 apart and start can't be higher than the end.\n";
    cout << "Digit 0 in both inputs to finish the program. \n\n";
    
    int again = main_code();
    while(again != 0){
        again = main_code();
    }
    cout << "Finishing...";
    return 0;
}

// >> MAIN CODE <<
int main_code(){
    int start, end;
//inputs
    while(true){
        cout << "The start: \n"; cin >> start;
        cout << "The end: \n"; cin >> end;
        if (start == end && end == 0){return 0;}
        else if(start > end){cout<< "Sorry, end must be higher than start\n\n";}
        else if (start == end){cout<< "Sorry, numbers must be different\n\n";}
        else if (start+1 == end){cout<< "Sorry, numbers must be at least 2 apart.\n\n";}
        else{break;}
    }

// Variables
    int numbs_between = nums_in(start, end);
    int sum_total = sum(start, end);
    float average = (double)sum_total / numbs_between;
    int negatives = negatives_num(start, end);
    int positives = positive_nums(start, end);
    int evens = even(start, end);
    int odds = numbs_between - evens;

//Outputs
    cout << "There are "<< numbs_between << " numbers between " << start << " and " << end << "\n";
    cout << "The sum of all of them is " << sum_total << "\n";
    cout << "Their average is "<< average << "\n";
    cout << "There are "<< negatives << " negative numbers and " << positives << " positives.\n";
    cout << evens << " are even, " << odds << " are odds\n\n";
    return 1;
}

File: Functions/nums_between.py
# Back to learning Python
# Starting with learning how to write functions

# Made in 16/01/2026, a re-make of "nums_between.cpp" from my 2025 cpp repository

def evens_counter(s, e): # returns how many evens between start and end
    evens_amount = 0
    for i in range(s+1, e):
        if(i%2 == 0):
            evens_amount += 1
    return evens_amount

def sum_counter(s, e): # returns the sum of all the numbers between
    total = 0
    for i in range(s+1, e):
        total += i
    return total

def negatives_counter(s, e):
    if (s>0):
        return 0
    else:
        negatives = 0
        for i in range(s+1, e):
            if(i < 0):
                negatives += 1
        return negatives

def main():
    start = int(input("Start: ")) #inputs
    end = int(input("End: "))
    if(start == 0 and end == 0): # Finishing check
        return False

    while(start > end or start+1 == end): #validation
        if(start > end):
            print("Start must be smaller than end.")
        elif(start+1 == end):
            print("Numbers must be at least 2 apart.")

        start = int(input("Start: "))
        end = int(input("End: "))

    total_between = end - start - 1
    total_sum = sum_counter(start, end)
    total_average = total_sum/total_between
    negatives_between = negatives_counter(start, end)
    positives_between = total_between - negatives_between
    evens_between = evens_counter(start, end)
    odds_between = total_between - evens_between

    print(f"Between {start} and {end}, there are {total_between} numbers")
    print(f"The sum of them all is {total_sum}")
    print(f"The average is {total_average}")
    print(f"There are {negatives_between} negatives and {positives_between} positives")
    print(f"{evens_between} are evens, {odds_between} are odds")
    return True



print("In Between Info")
print("Rules: start must be smaller than end")
print("Numbers must be at least 2 apart")
print("Type 0 in both inputs to finish the program.")

again = main() # looping the code and leaving when both are 0
while again:
    again = main()


